packages/flutter/lib/src/rendering/shifted_box.dart

```diff
@@ -279,15 +280,15 @@ abstract class RenderAligningShiftedBox extends RenderShiftedBox {
        _textDirection = textDirection,
        super(child);
 
+  /// The [Alignment] to use for aligning the child.
+  ///
+  /// This is the [alignment] resolved against [textDirection]. Subclasses should
+  /// use [resolvedAlignment] instead of [alignment] directly, for computing the
+  /// child's offset.
+  @protected
```

Comment:
Do you mean implement `RenderAligningShiftedBox.computeDryBaseline` and let `_RenderInputPadding` extend this class instead?

Reply:
not for `_RenderAppBarTitleBox` because it changes the constraints before passing it onto its child:
```dart
class _RenderAppBarTitleBox extends RenderAligningShiftedBox {
  @override
  double? computeDryBaseline(covariant BoxConstraints constraints, TextBaseline baseline) {
    final BoxConstraints innerConstraints = constraints.copyWith(maxHeight: double.infinity);
    ...
  }
```